File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& a, vector<int>& b) 
    {
        int n = a.size() ;
        int m = b.size() ;
        unordered_map<int,int > mp ;
        set<int> st ;
        for(auto & x : a) mp[x]++;
        for(auto x : b)
        {
            if(mp.find(x) != mp.end())
            {
                st.insert(x);
            }
        }
        vector<int> ans(st.begin() , st.end());
        return ans ;
        
        
    }
};

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        
        //creat HashSet ;
        HashSet<Integer> st = new HashSet() ;
        List<Integer > lst = new ArrayList() ;
        for(int x : nums1)
        {
            st.add(x); // 1 2  
        }
        
        int i = 0 ;
        for(int x : nums2)
        {
            if(st.contains(x)) // 2 , 2
            {
                lst.add(x);
                st.remove(x);
            }
        }
        
        int [] arr = new int [lst.size() ] ;
        for(int ii = 0 ; ii < lst.size() ; ii++)
        {
            arr[ii] = lst.get(ii) ;
        }
        return arr;
        
    }
}